### PR TITLE
hardening/808_add_method_get_for_all_subscriptions

### PR DESCRIPTION
**Pull request by pcoello25**
* Partially implements issue #808 
  * Update CHANGES_NEXT_RELEASE is not neccesary
* 100% Unit test passed: 
```
Results :
Tests run: 66,  Failures: 0, Errors: 0, Skipped: 0 [cygnus-common]
Tests run: 103, Failures: 0, Errors: 0, Skipped: 0 [cygnus-ngsi]
```
* e2e (unofficial) tests passed:
```
$ curl -X GET http://localhost:8081/v1/subscriptions?ngsi_version=2 -d '{"host":"orion.lab.fi-ware.org", "port":"1026", "ssl":"false", "xauthtoken":"cv78xc6asdjfkhas78xzcv786zxcadsf"}'
{"success":"true","result" : {{"result":[{"notification":{"lastNotification":"2015-07-28T22:23:30.00Z","timesSent":7126568376946155044,"http":{"url":"http:\/\/130.206.82.120:1026\/ngsi10\/notifyContext"},"attrs":[]},"expires":"2016-09-13T09:27:15.00Z","subject":{"condition":{"attrs":["chlorine"]},"entities":[{"id":"","type":"sevilla:fountain","idPattern":"Sevilla:FUENTES:"}]},"id":"54228e731860a3873d395d66","status":"active"},{"notification":{"lastNotification":"2015-07-28T22:24:33.00Z","timesSent":8122767998226748692,"http":{"url":"http:\/\/130.206.82.120:1026\/ngsi10\/notifyContext"},"attrs":[]},"expires":"2016-09-25T08:17:47.00Z","subject":{"condition":{"attrs":["TimeInstant"]},"entities":[{"id":"","type":"sevilla:fountain","idPattern":"Sevilla:FUENTES"}]},"id":"5432502b1860a3873d395e95","status":"active"},{"notification":{"lastNotification":"2015-07-28T22:24:53.00Z","timesSent":8786703474088345564,"http":{"url":"http:\/\/130.206.83.12:1026\/ngsi10\/notifyContext"},"attrs":[]},"expires":"2035-01-06T09:11:45.00Z","subject":{"condition":{"attrs":["TimeInstant"]},"entities":[{"id":"","type":"sevilla:fountain","idPattern":"Sevilla:FUENTES"}]},"id":"553614511860a3f429e3fd80","status":"active"},{"notification":{"lastNotification":"2015-07-28T22:25:01.00Z","timesSent":8785875184958230546,"http":{"url":"http:\/\/130.206.83.12:1026\/ngsi10\/notifyContext"},"attrs":[]},"expires":"2035-01-06T09:27:18.00Z","subject":{"condition":{"attrs":["TimeInstant","longitude","latitude"]},"entities":[{"id":"","type":"sevilla:fountain","idPattern":"Sevilla:FUENTES:"}]},"id":"553617f61860a3f429e3fd85","status":"active"},{"notification":{"lastNotification":"2016-04-11T13:24:19.00Z","timesSent":7918500433202332193,"http":{"url":"http:\/\/52.16.174.229:2700"},"attrs":[]},"expires":"2016-07-11T21:33:03.00Z","subject":{"condition":{"attrs":["sound"]},"entities":[{"id":"","type":"santander:sound","idPattern":"urn:smartsantander:testbed:.*"}]},"id":"55a2dd0f9a3bb06493b38fef","status":"active"},{"notification":{"lastNotification":"2016-04-11T13:24:19.00Z","timesSent":8793806529804560773,"http":{"url":"http:\/\/52.16.174.229:2735\/sound"},"attrs":[]},"expires":"2016-07-27T08:43:13.00Z","subject":{"condition":{"attrs":["sound"]},"entities":[{"id":"","type":"santander:sound","idPattern":"urn:smartsantander:testbed:.*"}]},"id":"55b740a1760b7a367c9c7bd1","status":"active"},{"notification":{"lastNotification":"2016-04-11T13:24:19.00Z","timesSent":8793414191782978685,"http":{"url":"http:\/\/lmctmlgw7thh.runscope.net\/sound"},"attrs":[]},"expires":"2016-07-27T08:46:15.00Z","subject":{"condition":{"attrs":["sound"]},"entities":[{"id":"","type":"santander:sound","idPattern":"urn:smartsantander:testbed:.*"}]},"id":"55b74157760b7a367c9c7bd2","status":"active"},{"notification":{"http":{"url":"http:\/\/130.206.123.223:5050\/notify"},"attrs":["taxiId","time","lat","lon","status"]},"expires":"2016-09-03T10:43:17.00Z","subject":{"condition":{"attrs":["time"]},"entities":[{"id":"","type":"Taxi","idPattern":".*"}]},"id":"55f00d453de71c949d1422e1","status":"active"},{"notification":{"http":{"url":"http:\/\/130.206.123.223:5050\/notify"},"attrs":["taxiId","time","lat","lon","status"]},"expires":"2016-09-03T11:22:52.00Z","throttling":1,"subject":{"condition":{"attrs":["time"]},"entities":[{"id":"","type":"Taxi","idPattern":".*"}]},"id":"55f0168c3de71c949d1422e2","status":"active"},{"notification":{"lastNotification":"2016-03-09T22:10:04.00Z","timesSent":8512673826141801965,"http":{"url":"http:\/\/130.206.123.223:5050\/notify"},"attrs":["taxiId","time","lat","lon","status"]},"expires":"2016-09-03T11:26:58.00Z","throttling":1,"subject":{"condition":{"attrs":["time"]},"entities":[{"id":"","type":"Taxi","idPattern":".*"}]},"id":"55f017823de71c949d1422e3","status":"active"},{"notification":{"lastNotification":"2015-09-24T14:56:08.00Z","timesSent":8040851773959392932,"http":{"url":"http:\/\/exasa.gr\/various\/test.php"},"attrs":[]},"expires":"2025-09-21T13:30:50.00Z","subject":{"condition":{"attrs":["status"]},"entities":[{"id":"40.64059722.944096","type":"location","idPattern":""}]},"id":"5603fb0aebf4aa5a1588cd21","status":"active"},{"notification":{"http":{"url":"http:\/\/exasa.gr\/various\/test.php"},"attrs":[]},"expires":"2025-09-21T14:10:26.00Z","subject":{"condition":{"attrs":["status"]},"entities":[{"id":"crowdId","type":"crowd","idPattern":""}]},"id":"56040452ebf4aa5a1588cd22","status":"active"},{"notification":{"lastNotification":"2015-09-24T14:11:30.00Z","timesSent":8040851773959392932,"http":{"url":"http:\/\/exasa.gr\/various\/test.php"},"attrs":[]},"expires":"2025-09-21T14:10:54.00Z","subject":{"condition":{"attrs":["observations"]},"entities":[{"id":"crowdId","type":"crowd","idPattern":""}]},"id":"5604046eebf4aa5a1588cd23","status":"active"},{"notification":{"lastNotification":"2015-09-24T14:24:04.00Z","timesSent":8040851773959392932,"http":{"url":"http:\/\/exasa.gr\/various\/test.php"},"attrs":[]},"expires":"2025-09-21T14:23:15.00Z","subject":{"condition":{"attrs":["status"]},"entities":[{"id":"40.64059781.944096","type":"location","idPattern":""}]},"id":"56040753ebf4aa5a1588cd24","status":"active"},{"notification":{"lastNotification":"2015-09-24T14:25:21.00Z","timesSent":7512234279060279666,"http":{"url":"http:\/\/exasa.gr\/various\/test.php"},"attrs":[]},"expires":"2025-09-21T14:25:20.00Z","subject":{"condition":{"attrs":["status"]},"entities":[{"id":"40.64059777777.944096","type":"location","idPattern":""}]},"id":"560407d0ebf4aa5a1588cd25","status":"active"},{"notification":{"lastNotification":"2015-09-24T15:12:36.00Z","timesSent":7512234279060279666,"http":{"url":"http:\/\/exasa.gr\/various\/test.php"},"attrs":[]},"expires":"2025-09-21T15:12:36.00Z","subject":{"condition":{"attrs":["status"]},"entities":[{"id":"40.64059712.944096","type":"location","idPattern":""}]},"id":"560412e4ebf4aa5a1588cd26","status":"active"},{"notification":{"lastNotification":"2015-09-25T12:44:22.00Z","timesSent":8231085825504083150,"http":{"url":"http:\/\/exasa.gr\/various\/test.php"},"attrs":[]},"expires":"2025-09-22T10:37:15.00Z","subject":{"condition":{"attrs":["aqiText"]},"entities":[{"id":"40.64368522.963004","type":"location","idPattern":""}]},"id":"560523dbf0ad42c03e86d912","status":"active"},{"notification":{"lastNotification":"2015-09-25T12:44:40.00Z","timesSent":7901856702247101528,"http":{"url":"http:\/\/exasa.gr\/various\/test.php"},"attrs":[]},"expires":"2025-09-22T10:42:07.00Z","subject":{"condition":{"attrs":["aqiText"]},"entities":[{"id":"40.60113922.960463","type":"location","idPattern":""}]},"id":"560524fff0ad42c03e86d918","status":"active"},{"notification":{"lastNotification":"2015-09-25T12:44:39.00Z","timesSent":6886708066761051270,"http":{"url":"http:\/\/exasa.gr\/various\/test.php"},"attrs":[]},"expires":"2025-09-22T10:43:10.00Z","subject":{"condition":{"attrs":["aqiText"]},"entities":[{"id":"40.64059922.944096","type":"location","idPattern":""}]},"id":"5605253ef0ad42c03e86d91b","status":"active"},{"notification":{"lastNotification":"2015-09-25T12:44:39.00Z","timesSent":8578622459237801700,"http":{"url":"http:\/\/exasa.gr\/various\/test.php"},"attrs":[]},"expires":"2025-09-22T10:43:14.00Z","subject":{"condition":{"attrs":["aqiText"]},"entities":[{"id":"40.62371522.957265","type":"location","idPattern":""}]},"id":"56052542f0ad42c03e86d91c","status":"active"}]}}
``` 
* Assignee: @frbattid

### CODE REVIEW

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/orion/OrionBackendImpl.java [hunk 1]
```diff
@@ -140,6 +140,17 @@ public JsonResponse deleteSubscriptionV2(String subscriptionId, String token) th
         
         return headers;
     }
+
+    @Override
+    public JsonResponse getSubscriptionsV2(String token, String subscriptionId) throws Exception {
+    // create the http header
```
Comment (frbattid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/orion/OrionBackendImpl.java]: Wrong alignment

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java [hunk 2]
```diff
@@ -436,27 +439,23 @@ protected void handleGetSubscriptions(HttpServletRequest request, HttpServletRes
...
         } // if
         
-        if (subscriptionID == null) {
-            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        if (ngsiVersion.equals("1")) {
+            response.setStatus(HttpServletResponse.SC_NOT_IMPLEMENTED);
             response.getWriter().println("{\"success\":\"false\","
-                + "\"error\":\"Parse error, missing parameter (subscription_id). Check it for errors.\"}");
-            LOGGER.error("Parse error, missing parameter (subscription_id). Check it for errors.");
+                + "\"error\":\"GET /v1/subscriptions not implemented.\"}");
```
Comment (frbattid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: I would add something like "... for NGSI version 1".

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java [hunk 3]
```diff
@@ -519,35 +522,69 @@ protected void handleGetSubscriptions(HttpServletRequest request, HttpServletRes
...
-            int status = -1;
-            JSONObject orionJson = new JSONObject();
+            try { 
+                int status = -1;
+                JSONObject orionJson = new JSONObject();
+
+                JsonResponse orionResponse = orionBackend.
+                getSubscriptionsV2(token, subscriptionID);
+                
+                if (orionResponse != null) {
```
Comment (frbattid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: What happens if `orionResponse == null`?

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java [hunk 4]
```diff
@@ -519,35 +522,69 @@ protected void handleGetSubscriptions(HttpServletRequest request, HttpServletRes
...
+            
+            try { 
+                int status = -1;
+                JSONObject orionJson = new JSONObject();
 
-            LOGGER.debug("Status code got: " + status);
+                JsonResponse orionResponse = orionBackend.
+                getSubscriptionsByIdV2(token, subscriptionID);
+
+                if (orionResponse != null) {
```
Comment (frbattid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: What happens if `orionResponse == null`?

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java (hunk 3 above)
Comment (pcoello25) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: Forgot to check that path. Now it's fixed in 8dd9f78.

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java (hunk 2 above)
Comment (pcoello25) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: Fixed in 8dd9f78

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/orion/OrionBackendImpl.java (hunk 1 above)
Comment (pcoello25) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/orion/OrionBackendImpl.java]: Fixed in 8dd9f78

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java (hunk 4 above)
Comment (pcoello25) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: Forgot to check that path. Now it's fixed in 1b3ab54

**Comment by frbattid** [comment]
LGTM

**Pull request merged**